lib/smart_proxy_reports/reports.rb

```diff
@@ -0,0 +1,29 @@
+module Proxy::Reports
+  class PluginConfiguration
+    def load_classes
+      require "smart_proxy_reports/spooled_http_client"
+    end
+
+    def load_dependency_injection_wirings(container, _settings)
+      container.singleton_dependency :reports_spool, -> {
+          SpooledHttpClient.instance.initialize_directory
+        }
+    end
+  end
+
+  class Plugin < ::Proxy::Plugin
+    plugin :reports, Proxy::Reports::VERSION
+
+    default_settings reported_proxy_hostname: "localhost",
```

Comment:
Also unrelated, but you can consider
```suggestion
    default_settings reported_proxy_hostname: Socket.gethostname(),
```

Reply:
Fixed.